python/lsst/sims/catUtils/supernovae/snObject.py

```diff
@@ -454,7 +473,8 @@ def SNObjectSED(self, time, wavelen=None, bandpass=None,
 
         # Set SED to 0 beyond the model phase range, will change this if
         # SNCosmo includes a more sensible decay later.
-        if (time > self.mintime()) & (time < self.maxtime()):
+        if (time > self.mintime()) & (time < self.maxtime())\
+                & (self.modelOutSideRange=='zero'):
```

Comment:
I don't understand this if statement.  The first argument (time>self.mintime() & time<self.maxtime()) guarantees that the model is inside the allowable time range.  Why do you need to test on self.modelOutSideRange?